### DISCUSSION TITLE
'rebalance' for mqtt-proxy via clientid

### DISCUSSION THREAD

**Discussion by liandimx** [apache/apisix#6233] [Ideas]
简单试验了下2.12发布的mqtt-proxy根据clientid选择发送对象的功能，如预期一样动作，赞！

不过如果集群中的某个mqtt broker（简称A）停掉后再启动，
那么这个A不会再被分配到原有clientid的任何请求了，
只会被分配到新产生clientid的请求。
虽说这也是可以解释的，毕竟原来的那些clientid已被重新分配到别的broker了，
但是这时A就会很空闲，而其他的broker负荷加重。
重启apisix会重新进行分配，但这种人为增加数据丢失风险的操作是不可取的。

所以想mqtt-proxy能否提供一个类似‘rebalance’的机能，
让用户不重启apisix，就能重新分配负荷？

**Comment by tzssangglass** [comment]
@spacewander What do you think about this

**Reply by spacewander** [reply]
Well, Nginx doesn't support reassigning clients to a different upstream.
